File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
         unordered_map<char,int>   mp;
      int l =0;
      int len =0;
      for(int i =0; i<s.size(); i++){
        mp[s[i]]++;
        while(mp[s[i]]>1){
            mp[s[l]]--;
            l++;
        }
        len= max(len, i-l+1);

      }
      return len;
        
    }
};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    maxDepth(root) {
        if(!root) return 0;
         let leftH = this.maxDepth(root.left);
         let rightR = this.maxDepth(root.right);
         let maxi = Math.max(leftH,rightR);
         console.log(maxi);
         return maxi+1;
    }
}
